config/src/test/java/beanconfig/ObjectsConfig.java

```diff
@@ -24,6 +26,14 @@ public String getOptionalValue() {
         public void setOptionalValue(String optionalValue) {
           this.optionalValue = optionalValue;
         }
+
+        public String getNullableValue() {
+            return nullableValue;
+        }
+
+        public void setNullableValue(String nullableValue) {
+            this.nullableValue = nullableValue;
```

Comment:
I think you're right. If an optional field is defaulted (either through an initializer or the default constructor), and if that field is missing in the configuration, then that default value should take effect. That does seem more intuitive to me.